frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/ModelManager.java

```diff
@@ -126,20 +126,38 @@ public ModelArchive registerModel(
         return archive;
     }
 
-    public boolean unregisterModel(String modelName) {
+    public HttpResponseStatus unregisterModel(String modelName) {
         Model model = models.remove(modelName);
         if (model == null) {
             logger.warn("Model not found: " + modelName);
-            return false;
+            return HttpResponseStatus.NOT_FOUND;
         }
-
+        logger.warn("setMinWorkers(0)");
         model.setMinWorkers(0);
+        logger.warn("setMaxWorkers(0)");
         model.setMaxWorkers(0);
-        wlm.modelChanged(model);
-        model.getModelArchive().clean();
-        startupModels.remove(modelName);
-        logger.info("Model {} unregistered.", modelName);
-        return true;
+        logger.warn("modelChanged(0)");
+        CompletableFuture<HttpResponseStatus> futureStatus = wlm.modelChanged(model);
+        HttpResponseStatus httpResponseStatus = HttpResponseStatus.OK;
+        try {
+            httpResponseStatus = futureStatus.get();
+        } catch (Exception e) {
```

Comment:
You are handling blanket exception here. Not a good idea. Handle specific exceptions.